.circleci/config.yml

```diff
@@ -61,36 +72,229 @@ jobs:
       - run:
           name: Lint Source Code
           command: make lint
+    
+  unit-test:
+    macos:
+      xcode: 13.4.0 # Specify the Xcode version to use
+
+    steps:
+      - checkout
+
+      # restore pods related caches
+      - restore_cache:
+          keys:
+            - 1-gems-{{ checksum "Gemfile.lock" }}
+
+      # make sure we're on the right version of cocoapods
+      - run:
+          name: Verify Cocoapods Version
+          command: bundle check || bundle install --path vendor/bundle
+
+      # save cocoapods version gem data
+      - save_cache:
+          key: 1-gems-{{ checksum "Gemfile.lock" }}
+          paths:
+            - vendor/bundle
+
+      # restore pods related caches
+      - restore_cache:
+          keys:
+            - cocoapods-cache-v1-{{ arch }}-{{ .Branch }}-{{ checksum "Podfile.lock" }}
+            - cocoapods-cache-v1-{{ arch }}-{{ .Branch }}
+            - cocoapods-cache-v1
+
+      # install CocoaPods - using default CocoaPods version, not the bundle
+      - run:
+          name: Repo Update & Install CocoaPods
+          command: make ci-pod-install
+
+      # save pods related files
+      - save_cache:
+          name: Saving CocoaPods Cache
+          key: cocoapods-cache-v1-{{ arch }}-{{ .Branch }}-{{ checksum "Podfile.lock" }}
+          paths:
+            - ./Pods
+            - ~/.cocoapods
```

Comment:
It looks like these steps to install CocoaPods and update the Pods are common to all the jobs. Can you look into using reusable configs in CircleCI or YAML anchors and aliases to define these steps once and reuse in each job?
Here are some references:
https://preventdirectaccess.com/reuse-steps-circleci/
CircleCI reusable config guide: https://circleci.com/docs/reusing-config/
YAML anchors and aliases: https://circleci.com/blog/circleci-hacks-reuse-yaml-in-your-circleci-config-with-yaml/